### DISCUSSION TITLE
Lift MSRV requirement

### DISCUSSION THREAD

**Discussion by flokli** [twosigma/nsncd#54] [Ideas]
As could be seen in the [previous PRs](https://github.com/twosigma/nsncd/pull/51#issuecomment-1321001752), the MSRV requirement is causing a lot of churn:

 - some crates are not available, or much worse than their current version
 - the UX to edit Cargo.toml "to downgrade sufficiently" is terrible
 - for an external contributor, this is only visible in CI or by building in some oldstable container locally 

I understand you need to still ship an nsncd binary to Debian oldstable, but this is causing a lot of pain.

Could you use a more recent rust compiler to produce it?
Or branch off the Debian 10 bits to a "maintenance branch" and only cherry-pick necessary fixes to there?

**Comment by leifwalsh** [comment]
@geofft what do you think? I think you're the only one of us qualified to speak to whether we could do this.
